crates/wasmtime/src/runtime/externals/table.rs

```diff
@@ -203,14 +208,14 @@ impl Table {
     /// # Panics
     ///
     /// Panics if `store` does not own this table.
-    pub fn set(&self, mut store: impl AsContextMut, index: u32, val: Ref) -> Result<()> {
+    pub fn set(&self, mut store: impl AsContextMut, index: u64, val: Ref) -> Result<()> {
         let store = store.as_context_mut().0;
         let ty = self.ty(&store);
         let val = val.into_table_element(store, ty.element())?;
         let table = self.wasmtime_table(store, iter::empty());
         unsafe {
             (*table)
-                .set(index, val)
+                .set(index.try_into().unwrap(), val)
```

Comment:
Similar to my comment above, could this cast's error be handled to return an error in this case? (it basically means out-of-bounds)